nntrainer/utils/base_properties.h

```diff
@@ -20,30 +24,63 @@
 
 namespace nntrainer {
 
+template <typename T>
+using remove_cv_ref_t = std::remove_cv_t<std::remove_reference_t<T>>;
 /**
- * @brief property tag to specialize functions based on this
+ * @brief property info to specialize functions based on this
  *
  * @tparam T property type
  */
-template <typename T> struct prop_tag { using type = typename T::prop_tag; };
+template <typename T> struct prop_info {
+  using prop_type = remove_cv_ref_t<T>;
```

Comment:
Add more comments here? what is prop_type, tag_type, data_type with example?

Reply:
Updated docs and added test code for clarity :)